### PR TITLE
Make corehost shim thinner

### PR DESCRIPTION
**Pull request by schellap**
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/1984)

### CODE REVIEW

**Comment by schellap** [comment]
@gkhanna79 a quick look perhaps...

File: src/corehost/cli/fxr/fx_ver.cpp [hunk 1]
```diff
@@ -60,22 +60,38 @@ pal::string_t fx_ver_t::as_str()
...
+
+    if (a.m_patch != b.m_patch)
+    {
+        return (a.m_patch > b.m_patch) ? 1 : -1;
+    }
+
+    if (a.m_pre.empty() != b.m_pre.empty())
+    {
+        // Either a is empty or b is empty
+        return a.m_pre.empty() ? 1 : -1;
```
Comment (gkhanna79) [inline src/corehost/cli/fxr/fx_ver.cpp]: What does it mean to return "-1" if a is empty but is not?

**Comment by gkhanna79** [comment]
LGTM

File: src/corehost/cli/fxr/fx_ver.cpp [hunk 2]
```diff
@@ -60,22 +60,38 @@ pal::string_t fx_ver_t::as_str()
...
-                ? ((a.m_pre.empty() == b.m_pre.empty())
-                    ? ((a.m_pre.empty())
-                        ? (ignore_build ? 0 : a.m_build.compare(b.m_build))
-                        : a.m_pre.compare(b.m_pre))
-                    : a.m_pre.empty() ? 1 : -1)
-                : (a.m_patch > b.m_patch ? 1 : -1))
-            : (a.m_minor > b.m_minor ? 1 : -1))
-        : ((a.m_major > b.m_major) ? 1 : -1)
-        ;
+    if (a.m_major != b.m_major)
```
Comment (krwq) [inline src/corehost/cli/fxr/fx_ver.cpp]: nice! that removed part looks a little scary

**Pull request merged**